agent/consul/state/config_entry.go

```diff
@@ -1435,6 +1441,28 @@ func readDiscoveryChainConfigEntriesTxn(
 		res.Services[svcID] = entry
 	}
 
+	peerEntMeta := structs.DefaultEnterpriseMetaInPartition(entMeta.PartitionOrDefault())
+	for peerName := range todoPeers {
+		q := Query{
+			Value:          strings.ToLower(peerName),
+			EnterpriseMeta: *peerEntMeta,
+		}
+		idx, entry, err := peeringReadTxn(tx, ws, q)
+		if err != nil {
+			return 0, nil, err
+		}
+		if idx > maxIdx {
+			maxIdx = idx
+		}
+
+		if entry == nil {
+			res.Peers[peerName] = nil
```

Comment:
Is there a need for this if-statement? I think it's equivalent to just `res.Peers[peerName] = entry`